File: 75-sort-colors/75-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
         int zeros = 0;
        int ones = 0;
        int twos = 0;
        for (int i : nums) {
            if (i == 0) zeros++;
            if (i == 1) ones++;
            if (i == 2) twos++;
        }
        for (int i = 0; i < zeros; i++) {
            nums[i] = 0;
        }
        for (int i = zeros; i < zeros+ones; i++) {
            nums[i] = 1;
        }
        for (int i = zeros+ones; i < zeros+ones+twos; i++) {
            nums[i] = 2;
        }
    }
}

File: 75-sort-colors/75-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
       
 int a=0,b=nums.size()-1,i=0;
        
        while(i<=b)
        {
            if(nums[i]==2)
            {
                swap(nums[i],nums[b]);
                b--;
            }
            else if(nums[i]==0)
            {
                swap(nums[i],nums[a]);
                i=a+1;
                a++;
            }
           else
               i++;
        }
    }
};